### DISCUSSION TITLE
formily/antd 没有 tree-shaking？

### DISCUSSION THREAD

**Discussion by liuweiGL** [alibaba/formily#1514] [Q&A 2.x] [answered]
## 复现仓库

https://github.com/liuweiGL/formily-not-tree-shaking

## 复现步骤

1. clone repo
2. yarn
3. yarn build
4. 查看 dist/assets 下的 formily_antd.js 跟 lodash.js，可以发现 lodash 是进行 tree-shaking 的，但是 formily/antd 没有。

**Comment by liuweiGL** [comment]
另外一个问题：vite使用 rollup 打包，支持 esm 模块的 tree-shaking，所以不会使用 babel-import-plugin 这样的插件进行**代码层面**的按需加载，而仅仅是使用 [vite-plugin-style-import](https://github.com/anncwb/vite-plugin-style-import/blob/main/src/index.ts) 进行**样式层面**的按需加载。
所以在 @formily/antd 根目录中的 index.ts 中 import style.ts 会导致整个样式被导入，具体可查看 dist/assets/formily_antd.css 文件。

**Comment by liuweiGL** [comment]
其他：
1. https://github.com/ant-design/babel-plugin-import/issues/325#issuecomment-600408841
2. https://github.com/vitejs/vite/issues/1389#issuecomment-762139761

**Comment by janryWang** [comment, marked as answer]
暂时不考虑支持摇树，推荐用babel-import-plugin，因为两者方案有冲突

**Comment by liuweiGL** [comment]
类库支持摇树特性需要满足什么规范吗？后面我可以研究一下。
